controller/src/main/java/org/jboss/as/controller/CapabilityReferenceRecorder.java

```diff
@@ -76,6 +76,21 @@ default boolean isDynamicDependent() {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * dynamicElements the dynmaic elements of the requirement capability name.
+     * @return requirement name of the capability this reference depends on
+     */
+    default String getRequirementName(String... dynamicElements) {
+        if (dynamicElements != null && dynamicElements.length > 0) {
+            String[] dynamicParts = new String[dynamicElements.length];
+            for (int i = 0; i < dynamicElements.length; i++) {
+                dynamicParts[i] = "$" + dynamicElements[i];
+            }
+            return RuntimeCapability.buildDynamicCapabilityName(getBaseRequirementName(), dynamicParts);
```

Comment:
It's better to not repeat the static part. We've gotten complaints before about the size of our metadata.

Reply:
ok